azure/services/managedclusters/spec.go

```diff
@@ -0,0 +1,430 @@
+	// Name is the name of this AKS Cluster.
+	Name string
+
+	// ResourceGroup is the name of the Azure resource group for this AKS Cluster.
+	ResourceGroup string
+
+	// NodeResourceGroup is the name of the Azure resource group containing IaaS VMs.
+	NodeResourceGroup string
+
+	// VnetSubnetID is the Azure Resource ID for the subnet which should contain nodes.
+	VnetSubnetID string
+
+	// Location is a string matching one of the canonical Azure region names. Examples: "westus2", "eastus".
+	Location string
+
+	// Tags is a set of tags to add to this cluster.
+	Tags map[string]string
+
+	// Version defines the desired Kubernetes version.
+	Version string
+
+	// LoadBalancerSKU for the managed cluster. Possible values include: 'Standard', 'Basic'. Defaults to Standard.
+	LoadBalancerSKU string
+
+	// NetworkPlugin used for building Kubernetes network. Possible values include: 'azure', 'kubenet'. Defaults to azure.
+	NetworkPlugin string
+
+	// NetworkPolicy used for building Kubernetes network. Possible values include: 'calico', 'azure'. Defaults to azure.
+	NetworkPolicy string
+
+	// SSHPublicKey is a string literal containing an ssh public key. Will autogenerate and discard if not provided.
+	SSHPublicKey string
+
+	// GetAllAgentPools is a function that returns the list of agent pool specifications in this cluster.
+	GetAllAgentPools func() ([]azure.AgentPoolSpec, error)
+
+	// PodCIDR is the CIDR block for IP addresses distributed to pods
+	PodCIDR string
+
+	// ServiceCIDR is the CIDR block for IP addresses distributed to services
+	ServiceCIDR string
+
+	// DNSServiceIP is an IP address assigned to the Kubernetes DNS service
+	DNSServiceIP *string
+
+	// AddonProfiles are the profiles of managed cluster add-on.
+	AddonProfiles []AddonProfile
+
+	// AADProfile is Azure Active Directory configuration to integrate with AKS, for aad authentication.
+	AADProfile *AADProfile
+
+	// SKU is the SKU of the AKS to be provisioned.
+	SKU *SKU
+
+	// LoadBalancerProfile is the profile of the cluster load balancer.
+	LoadBalancerProfile *LoadBalancerProfile
+
+	// APIServerAccessProfile is the access profile for AKS API server.
+	APIServerAccessProfile *APIServerAccessProfile
+
+	// Headers is the list of headers to add to the HTTP requests to update this resource.
+	Headers map[string]string
+}
+
+// AADProfile is Azure Active Directory configuration to integrate with AKS, for aad authentication.
+type AADProfile struct {
+	// Managed - Whether to enable managed AAD.
+	Managed bool
+
+	// EnableAzureRBAC - Whether to enable Azure RBAC for Kubernetes authorization.
+	EnableAzureRBAC bool
+
+	// AdminGroupObjectIDs - AAD group object IDs that will have admin role of the cluster.
+	AdminGroupObjectIDs []string
+}
+
+// AddonProfile - Profile of managed cluster add-on.
+type AddonProfile struct {
+	Name    string
+	Config  map[string]string
+	Enabled bool
+}
+
+// SKU - AKS SKU.
+type SKU struct {
+	// Tier - Tier of a managed cluster SKU.
```

Comment:
I'm pedantic enough to flag all these, but I prefer complete sentences for doc comments.

Reply:
I agree - I have to admit I just lift and shifted those as-is from types.go. Will cleanup the comments.